extensions/azurecore/src/utils/Logger.ts

```diff
@@ -6,19 +6,52 @@
 import * as vscode from 'vscode';
 import * as loc from '../localizedConstants';
 
+export enum LogLevel {
+	'All',
+	'Off',
+	'Critical',
+	'Error',
+	'Warning',
+	'Information',
+	'Verbose',
+}
+
 export class Logger {
 
-	public static channel: vscode.OutputChannel = vscode.window.createOutputChannel(loc.extensionName);
 
+	public static channel: vscode.OutputChannel = vscode.window.createOutputChannel(loc.extensionName);
 	private static _piiLogging: boolean = false;
+	public static config = vscode.workspace.getConfiguration('mssql');
 
-	static log(msg: any, ...vals: any[]) {
-		const fullMessage = `${msg} - ${vals.map(v => JSON.stringify(v)).join(' - ')}`;
-		this.channel.appendLine(fullMessage);
+	static write(logLevel: LogLevel, msg: any, ...vals: any[]) {
+		switch (logLevel) {
+			case LogLevel.Error:
+				if (LogLevel[logLevel] === this.config.tracingLevel || this.config.tracingLevel === 'Verbose') {
+					const fullMessage = `[${LogLevel[logLevel]}]: ${msg} - ${vals.map(v => JSON.stringify(v)).join(' - ')}`;
+					this.channel.appendLine(fullMessage);
+				}
+				break;
+			case LogLevel.Critical:
+				if (LogLevel[logLevel] === this.config.tracingLevel || this.config.tracingLevel === 'Verbose') {
+					const fullMessage = `[${LogLevel[logLevel]}]: ${msg} - ${vals.map(v => JSON.stringify(v)).join(' - ')}`;
+					this.channel.appendLine(fullMessage);
+				}
+				break;
+			case LogLevel.All:
+			case LogLevel.Off:
+			case LogLevel.Warning:
+			case LogLevel.Information:
+			case LogLevel.Verbose:
+				if (LogLevel[logLevel] === this.config.tracingLevel || this.config.tracingLevel === 'Verbose') {
+					const fullMessage = `[${LogLevel[logLevel]}]: ${msg} - ${vals.map(v => JSON.stringify(v)).join(' - ')}`;
+					this.channel.appendLine(fullMessage);
+				}
+				break;
+		}
 	}
 
 	static error(msg: any, ...vals: any[]) {
```

Comment:
This doesn't appear to be using the verbosity filter you added. It should really just be a thin wrapper over the `write` function IMO